benchmark/multidimensional_memset_benchmark_kernels.cpp

```diff
@@ -0,0 +1,24 @@
+
+// Copyright (c) 2023 NVIDIA Corporation (reply-to: brycelelbach@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <flux/op/cartesian_product.hpp>
+#include <flux/source/iota.hpp>
+#include <flux/op/for_each.hpp>
+
+void memset_2d_reference(double* A, std::size_t N, std::size_t M)
+{
+    for (std::size_t j = 0; j != M; ++j)
+        for (std::size_t i = 0; i != N; ++i)
+            A[i + j * N] = 0.0;
+}
+
+void memset_2d_flux_cartesian_product_iota(double* A, std::size_t N, std::size_t M)
+{
+    flux::cartesian_product(flux::iota(0LU, N), flux::iota(0LU, M))
+        .for_each(flux::unpack([&] (auto i, auto j) {
+            A[i + j * N] = 0.0;
+        }));
+}
```

Comment:
To match the reference version above, this should be

```suggestion
void memset_2d_flux_cartesian_product_iota(double* A, std::size_t N, std::size_t M)
{
    flux::cartesian_product(flux::iota(0LU, M), flux::iota(0LU, N))
        .for_each(flux::unpack([&] (auto j, auto i) {
            A[i + j * N] = 0.0;
        }));
}
```

which (for me) gives the same runtime for both

Reply:
Fixed slightly differently in 4132087e2b68de4ad69c0643e1cbaff2b67b5ecd - I switched the iteration order of the reference, instead of switching the iteration order of the flux variant.